### ISSUE TITLE
Enhancement: [prefer-destructuring] allow disabling for properties

### ISSUE THREAD

**Issue by rentalhost** [typescript-eslint/typescript-eslint#10515]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-destructuring/

### Description

Preferring destructuring is a great rule, however, it makes the code quite confusing if strictly applied to objects. Therefore, it would be interesting to have a rule where we could suggest it only for simple variables.

### Cases

```TypeScript
const user = { name: { first: "John", name: "Doe" } };
const user2 = { name: { first: "Mary", name: "Doe" } }

// Case #1
// Now: use object destructuring.
const firstName1 = user.name.first;

// Now: solution, but bad to reading.
const {
  name: { first: firstName2 },
} = user;

// Suggestion: do not report to more than 2 or more depth property.
const firstName3 = user.name.first;

// Case #2
// Now: use object destructuring.
user2.name = user.name;

// Now: no solution.
// Suggestion: do not report when assignment to is not a variable.
```

**Comment by Josh-Cena** [comment]
Thanks for the suggestion. `prefer-destructuring` is a core ESLint rule, and your suggestion does not involve any TS specific syntax, so it should be proposed to ESLint instead.
